packages/devextreme/js/__internal/core/di/index.ts

```diff
@@ -67,15 +73,37 @@ export class DIContext {
 
     const fabric = this.fabrics.get(id);
     if (fabric) {
-      const res: T = this.create(fabric as any);
-      this.instances.set(id, res);
-      this.instances.set(fabric, res);
-      return res;
+      const instance: T = this.create(fabric as any);
+
+      const decoratedInstance = this.applyGlobalDecorators(instance);
+
+      this.instances.set(id, decoratedInstance);
+      this.instances.set(fabric, decoratedInstance);
+      return instance;
     }
 
     return null;
   }
 
+  public decorator<T>(decoratorFn: DecoratorFunction<T>): void {
```

Comment:
Let's rename this method to `registerDecorator`
```suggestion
  public registerDecorator<T>(decoratorFn: DecoratorFunction<T>): void {
```

Reply:
Changed